src/request.rs

```diff
@@ -17,6 +17,9 @@ use crate::rmap::ResourceMap;
 #[derive(Clone)]
 /// An HTTP Request
 pub struct HttpRequest(pub(crate) Rc<HttpRequestInner>);
+// *. Rc<HttpRequestInner> is used exclusively and NO Weak<HttpRequestInner>
+// is allowed anywhere in the code. Weak pointer is purposely ignored when
+// doing Rc's ref counter check.
```

Comment:
make this a proper doc comment on HttpRequestInner

Reply:
HttpRequestInner type does not know the outer Rc wrapper. 
I feel the comment should be at HttpRequest level

Reply:
yep i like this too